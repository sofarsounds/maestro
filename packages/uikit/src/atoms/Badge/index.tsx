import styled, { css } from '../../lib/styledComponents';

interface BadgeProps {
  children: any;
  colour?: 'primary';
  clickable?: boolean;
  disabled?: boolean;
}

const Badge = styled.div<BadgeProps>`
  ${({ theme, colour, clickable }) => css`
    width: 36px;
    height: 36px;
    font-size: ${theme.fontSizes.body2};
    transition: ${theme.utils.transition()};
    border: none;
    border-radius: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    outline: none;
    user-select: none;

    ${clickable &&
      css`
        cursor: pointer;

        &:hover {
          background-color: rgba(17, 17, 17, 0.1);
        }
      `}

    ${colour === 'primary' &&
      css`
        background-color: ${theme.colours.primary};
        color: ${theme.colours.whiteDenim};
        font-weight: 600;

        ${clickable &&
          css`
            &:hover {
              background-color: ${theme.colours.green800};
            }
          `}
      `}
  `}
`;

export default Badge;
