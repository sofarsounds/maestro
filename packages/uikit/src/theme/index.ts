import { css } from '../lib/styledComponents';

import ITheme from '../typings/theme';

const breakPoints = {
  phone: 0,
  tablet: 768,
  desktop: 992
};

const theme: ITheme = {
  colours: {
    primary: '#008736',

    // blacks & greys
    black: '#000000',
    backToBlack: '#111111',
    blackBetty: '#1E1E1E',
    paintItBlack: '#333333',
    blueSmoke: '#949494',
    macyGrey: '#DBDBDB',
    silverSprings: '#F5F5F5',
    whiteDenim: '#FFFFFF',

    // green variations
    green900: '#006721',
    green700: '#039842',
    green600: '#10AD52',
    green500: '#23BA57',
    green400: '#4FC571',
    green300: '#72D08A',
    green200: '#9DDCAC',
    green100: '#CFEACC',
    green50: '#E6F7EA',

    // additional
    redRedWine: '#EF4423'
  },

  fonts: {
    fontFamily: 'Open Sans'
  },
  fontSizes: {
    base: '16px',
    h1: '96px',
    h2: '60px',
    h3: '48px',
    h4: '34px',
    h5: '24px',
    h6: '20px',
    title: '74px',
    subtitle1: '16px',
    subtitle2: '14px',
    body1: '16px',
    body2: '14px',
    overline: '10px',
    caption: '10px',
    button: '14px'
  },

  media: Object.keys(breakPoints).reduce((acc, label: string) => {
    acc[label] = (...args: any) => css`
      @media (min-width: ${breakPoints[label]}px) {
        ${css.call(undefined, ...args)};
      }
    `;
    return acc;
  }, {}),

  borderRadius: {
    button: '4px'
  },

  utils: {
    transition: (attr?: string) => `${attr || 'all'} .2s ease`
  }
};

export default theme;
