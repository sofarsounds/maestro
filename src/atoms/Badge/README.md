# Badge

To implement `Badge` into your project you'll need to add this import
```js
import { Badge } from '@sofarsounds/maestro'
```

After adding the import you can use it simply like this
```html
<Badge>12</Badge>
```

## Props
Table below contains all types of props available in the Badge component

| Name          | Type          | Default         | Description                      |
| :------------ | :-----        | :-------------- | :------------------------------- |
| **children**  | `React.Node`  |                 | The content displayed in the badge
| colour        | [Enum](#enum) |                 | Optional prop to specifiy a colour. Otherwise background is transparent
| clickable     | `Boolean`     | `false`         | Whether the badge should have a cursor pointer and hover state
| disabled      | `Boolean`     | `false`         | Whether the prop has a disabled state
| data-qaid     | `string`      |                 | Optional prop for testing purposes

### Enum
| type    |
| :-------|
| primary |
