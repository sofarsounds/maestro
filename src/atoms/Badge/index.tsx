import styled, { css } from '../../lib/styledComponents';

interface BadgeProps {
  children: any;
  color?: 'primary';
  clickable?: boolean;
  disabled?: boolean;
}

const Badge = styled.div<BadgeProps>`
  ${({ theme, color, disabled, clickable }) => css`
    width: 36px;
    height: 36px;
    margin: 0;
    padding: 0;
    font-size: ${theme.fontSizes.body2};
    transition: ${theme.utils.transition()};
    border: none;
    border-radius: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    outline: none;
    user-select: none;



    ${clickable &&
      css`
        cursor: pointer;

        &:hover {
          background-color: rgba(17, 17, 17, 0.1);
        }
      `}

    ${color === 'primary' &&
      css`
        background-color: ${theme.colors.primary};
        color: ${theme.colors.whiteDenim};
        font-weight: 600;

        ${clickable &&
          css`
            &:hover {
              background-color: ${theme.colors.green800};
            }
          `}
      `}

    ${disabled &&
      css`
        user-select: none;
        cursor: not-allowed;

        &:hover {
          background-color: transparent;
        }
      `}
  `}
`;

export default Badge;
