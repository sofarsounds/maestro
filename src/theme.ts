import Color from 'color';

export default {
  colors: {
    primary: '#3498db',
    secondary: '#8e44ad',
    white: '#ffffff'
  },
  utils: {
    borderRadius: '4px',
    transition: () => `all .25s ease`,
    darken: (hex: string, factor: number = 1) =>
      Color(hex)
        .darken(0.2 * factor)
        .hex()
  }
};
