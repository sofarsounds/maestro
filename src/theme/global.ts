import { createGlobalStyle, css } from '../lib/styledComponents';

import IconFontTTF from '../assets/fonts/sofarsounds-icon-font.ttf';
import IconFontWOFF from '../assets/fonts/sofarsounds-icon-font.woff';
import IconFontSVG from '../assets/fonts/sofarsounds-icon-font.svg';

export default createGlobalStyle`
  ${({ theme }) => css`
    /* open-sans-regular - latin */
    @font-face {
      font-family: 'Open Sans';
      font-style: normal;
      font-weight: 400;
      src: url('../assets/fonts/open-sans-v26-latin-regular.eot'); /* IE9 Compat Modes */
      src: local(''),
           url('../assets/fonts/open-sans-v26-latin-regular.eot?#iefix') format('embedded-opentype'), /* IE6-IE8 */
           url('../assets/fonts/open-sans-v26-latin-regular.woff2') format('woff2'), /* Super Modern Browsers */
           url('../assets/fonts/open-sans-v26-latin-regular.woff') format('woff'), /* Modern Browsers */
           url('../assets/fonts/open-sans-v26-latin-regular.ttf') format('truetype'), /* Safari, Android, iOS */
           url('../assets/fonts/open-sans-v26-latin-regular.svg#OpenSans') format('svg'); /* Legacy iOS */
    }
    /* open-sans-600 - latin */
    @font-face {
      font-family: 'Open Sans';
      font-style: normal;
      font-weight: 600;
      src: url('../assets/fonts/open-sans-v26-latin-600.eot'); /* IE9 Compat Modes */
      src: local(''),
           url('../assets/fonts/open-sans-v26-latin-600.eot?#iefix') format('embedded-opentype'), /* IE6-IE8 */
           url('../assets/fonts/open-sans-v26-latin-600.woff2') format('woff2'), /* Super Modern Browsers */
           url('../assets/fonts/open-sans-v26-latin-600.woff') format('woff'), /* Modern Browsers */
           url('../assets/fonts/open-sans-v26-latin-600.ttf') format('truetype'), /* Safari, Android, iOS */
           url('../assets/fonts/open-sans-v26-latin-600.svg#OpenSans') format('svg'); /* Legacy iOS */
    }
    /* open-sans-700 - latin */
    @font-face {
      font-family: 'Open Sans';
      font-style: normal;
      font-weight: 700;
      src: url('../assets/fonts/open-sans-v26-latin-700.eot'); /* IE9 Compat Modes */
      src: local(''),
           url('../assets/fonts/open-sans-v26-latin-700.eot?#iefix') format('embedded-opentype'), /* IE6-IE8 */
           url('../assets/fonts/open-sans-v26-latin-700.woff2') format('woff2'), /* Super Modern Browsers */
           url('../assets/fonts/open-sans-v26-latin-700.woff') format('woff'), /* Modern Browsers */
           url('../assets/fonts/open-sans-v26-latin-700.ttf') format('truetype'), /* Safari, Android, iOS */
           url('../assets/fonts/open-sans-v26-latin-700.svg#OpenSans') format('svg'); /* Legacy iOS */
    }
    /* open-sans-800 - latin */
    @font-face {
      font-family: 'Open Sans';
      font-style: normal;
      font-weight: 800;
      src: url('../assets/fonts/open-sans-v26-latin-800.eot'); /* IE9 Compat Modes */
      src: local(''),
           url('../assets/fonts/open-sans-v26-latin-800.eot?#iefix') format('embedded-opentype'), /* IE6-IE8 */
           url('../assets/fonts/open-sans-v26-latin-800.woff2') format('woff2'), /* Super Modern Browsers */
           url('../assets/fonts/open-sans-v26-latin-800.woff') format('woff'), /* Modern Browsers */
           url('../assets/fonts/open-sans-v26-latin-800.ttf') format('truetype'), /* Safari, Android, iOS */
           url('../assets/fonts/open-sans-v26-latin-800.svg#OpenSans') format('svg'); /* Legacy iOS */
    }

    @font-face {
      font-family: 'sofarsounds-icon-font';
      src:
        url(${IconFontTTF}) format('truetype'),
        url(${IconFontWOFF}) format('woff'),
        url(${IconFontSVG}) format('svg');
      font-weight: normal;
      font-style: normal;
    }

    body,
    html,
    * {
      font-family: "${theme.fonts.regular}", sans-serif;
    }

    body {
      font-size: ${theme.fontSizes.base};
      color: ${theme.colors.backToBlack};
      margin: 0px;
    }

    html {
      box-sizing: border-box;
    }

    *,
    ::after,
    ::before {
      box-sizing: inherit;
    }

    a {
      color: ${theme.colors.backToBlack};
    }

    a:link, a:visited {
      text-decoration: none;
    }

    button {
      background: none;
      margin-left: 0px;
      margin-right: 0px;
    }
  `}
`;
