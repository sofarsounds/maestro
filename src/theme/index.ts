import { css } from '../lib/styledComponents';

import Theme from '../typings/theme';

// breakpoints define the minimum width of a device (>=)
export const breakPoints = {
  xs: 0,
  sm: 576,
  md: 768,
  lg: 992,
  xl: 1200
};

const theme: Theme = {
  colors: {
    primary: '#039842',

    // blacks & greys
    black: '#000000',
    backToBlack: '#111111',
    blackBetty: '#1E1E1E',
    paintItBlack: '#333333',
    blueSmoke: '#949494',
    macyGrey: '#DBDBDB',
    silverSprings: '#F5F5F5',
    whiteDenim: '#FFFFFF',

    // green variations
    green900: '#006721',
    green800: '#008736',
    green700: '#039842',
    green600: '#10AD52',
    green500: '#23BA57',
    green400: '#4FC571',
    green300: '#72D08A',
    green200: '#9DDCAC',
    green100: '#CFEACC',
    green50: '#E6F7EA',

    // additional
    redRedWine: '#EF4423'
  },

  fonts: {
    regular: 'Open Sans'
  },
  fontSizes: {
    base: '16px',
    title: { lg: '54px', md: '38px', xs: '26px' },
    h1: { lg: '48px', md: '34px', xs: '24px' },
    h2: { lg: '34px', md: '24px', xs: '20px' },
    h3: { lg: '24px', md: '20px', xs: '16px' },
    h4: { lg: '20px', md: '16px', xs: '16px' },
    h5: { lg: '16px', md: '14px', xs: '14px' },
    h6: { lg: '14px', md: '14px', xs: '14px' },
    body1: '16px',
    body2: '14px',
    overline: '10px',
    caption: '10px',
    button: '14px',
    formGroupLabel: '14px'
  },

  media: Object.keys(breakPoints).reduce((acc, label: string) => {
    acc[label] = (...args: any) => css`
      @media (min-width: ${breakPoints[label]}px) {
        ${css.call(undefined, ...args)};
      }
    `;
    return acc;
  }, {}),

  ruler: Array.from({ length: 21 }, (e, i) => i * 4),
  zIndex: {
    navbar: 100
  },

  dimensions: {
    inputHeight: '44px',
    buttonHeight: '44px',
    radioBox: '20px',
    containerWidth: {
      small: '600px',
      default: '800px',
      large: '1000px',
      fullscreen: '100%'
    },

    button: {
      small: {
        height: '40px',
        padding: '0 15px'
      },

      default: {
        height: '44px',
        padding: '0 25px'
      }
    },

    navbarHeight: {
      lg: '84px',
      md: '74px',
      xs: '62px'
    }
  },

  borderRadius: {
    default: '2px',
    large: '8px',
    button: '4px',
    input: '2px'
  },

  utils: {
    transition: (attr?: string) => `${attr || 'all'} .2s ease`
  }
};

export default theme;
