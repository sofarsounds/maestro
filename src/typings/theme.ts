/**
 * Main Theme File
 * This is where all variables and colors are defined that can
 * be accessed using props in every component
 */

interface FontSizeResponsive {
  xs: string;
  md: string;
  lg: string;
}

export type Colors =
  | 'primary'
  | 'black'
  | 'blackBetty'
  | 'paintItBlack'
  | 'darkSmoke'
  | 'blueSmoke'
  | 'lightBlueSmoke'
  | 'macyGrey'
  | 'darkGrey'
  | 'silverSprings'
  | 'whiteDenim'
  | 'green900'
  | 'green800'
  | 'green700'
  | 'green600'
  | 'green500'
  | 'green400'
  | 'green300'
  | 'green200'
  | 'green100'
  | 'green50'
  | 'redRedWine'
  | 'greenOnions'
  | 'greenGarden'
  | 'blueChristmas'
  | 'purpleHaze'
  | 'purplePills'
  | 'checkBerry'
  | 'blueMagic'
  | 'brownSugar'
  | 'pinkTriangle'
  | 'royalBlood'
  | 'peaches'
  | 'orangeCrush'
  | 'mellowYellow'
  | 'yellowSubmarine';

export default interface Theme {
  colors: {
    primary: string;

    // blacks & greys
    black: string;
    backToBlack: string;
    blackBetty: string;
    paintItBlack: string;
    darkSmoke: string;
    blueSmoke: string;
    lightBlueSmoke: string;
    macyGrey: string;
    darkGrey: string;
    silverSprings: string;
    whiteDenim: string;

    // green variations
    green900: string;
    green800: string;
    green700: string;
    green600: string;
    green500: string;
    green400: string;
    green300: string;
    green200: string;
    green100: string;
    green50: string;

    // additional
    greenOnions: string;
    greenGarden: string;
    purpleHaze: string;
    blueChristmas: string;
    purplePills: string;
    checkBerry: string;
    blueMagic: string;
    brownSugar: string;
    pinkTriangle: string;
    royalBlood: string;
    redRedWine: string;
    peaches: string;
    orangeCrush: string;
    mellowYellow: string;
    yellowSubmarine: string;
  };

  fonts: {
    regular: string;
  };
  fontSizes: {
    base: string;
    h1: FontSizeResponsive;
    h2: FontSizeResponsive;
    h3: FontSizeResponsive;
    h4: FontSizeResponsive;
    h5: FontSizeResponsive;
    h6: FontSizeResponsive;
    title: FontSizeResponsive;
    body1: string;
    body2: string;
    overline: string;
    caption: string;
    button: string;
    formGroupLabel: string;
  };

  media: any;

  ruler: number[];
  zIndex: {
    navbar: number;
  };

  dimensions: {
    inputHeight: string;
    buttonHeight: string;
    radioBox: string;
    containerWidth: {
      small: string;
      default: string;
      large: string;
      fullscreen: string;
    };

    button: {
      small: {
        height: string;
        padding: string;
      };

      default: {
        height: string;
        padding: string;
      };
    };
    navbarHeight: {
      lg: string;
      md: string;
      xs: string;
    };
  };

  borderRadius: {
    default: string;
    large: string;
    button: string;
    input: string;
  };
  utils: {
    transition: (a?: string) => string;
  };
}
