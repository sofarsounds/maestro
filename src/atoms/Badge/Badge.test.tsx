import React from 'react';
import { mountWithTheme } from '../../test';

import theme from '../../theme';
import Badge from './index';

describe('<Badge />', () => {
  const setup = (props: any) => mountWithTheme(<Badge {...props}>1</Badge>);

  describe('Default Badge', () => {
    it('renders the badge correctly', () => {
      const wrapper = setup({});

      expect(wrapper).toHaveStyleRule('width', '36px');
      expect(wrapper).toHaveStyleRule('height', '36px');
      expect(wrapper).toHaveStyleRule('border-radius', '100%');

      expect(wrapper).toMatchSnapshot();
    });

    it('is clickable', () => {
      const wrapper = setup({ clickable: true });

      expect(wrapper).toHaveStyleRule('cursor', 'pointer');
      expect(wrapper).toHaveStyleRule(
        'background-color',
        'rgba(17,17,17,0.1)',
        {
          modifier: ':hover'
        }
      );
    });
  });

  describe('Primary Badge', () => {
    it('renders the badge correctly', () => {
      const wrapper = setup({ colour: 'primary' });

      expect(wrapper).toHaveStyleRule(
        'background-color',
        theme.colours.primary
      );
      expect(wrapper).toHaveStyleRule('color', theme.colours.whiteDenim);
      expect(wrapper).toHaveStyleRule('font-weight', '600');

      expect(wrapper).toMatchSnapshot();
    });

    it('is clickable', () => {
      const wrapper = setup({ colour: 'primary', clickable: true });

      expect(wrapper).toHaveStyleRule(
        'background-color',
        theme.colours.green800,
        {
          modifier: ':hover'
        }
      );
    });
  });
});
