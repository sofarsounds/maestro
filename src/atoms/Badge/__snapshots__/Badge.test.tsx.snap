// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> Default Badge renders the badge correctly 1`] = `
.c0 {
  width: 36px;
  height: 36px;
  margin: 0;
  padding: 0;
  font-size: 14px;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  border: none;
  border-radius: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<Component
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "default": "2px",
        "input": "2px",
        "large": "8px",
      },
      "colors": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueChristmas": "#3498DB",
        "blueMagic": "#34495E",
        "blueSmoke": "#949494",
        "brownSugar": "#9B525F",
        "checkBerry": "#98035A",
        "darkSmoke": "#767676",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green800": "#008736",
        "green900": "#006721",
        "greenGarden": "#19BC9B",
        "greenOnions": "#8BC34B",
        "macyGrey": "#DBDBDB",
        "mellowYellow": "#F1C410",
        "orangeCrush": "#F39C12",
        "paintItBlack": "#333333",
        "peaches": "#F99973",
        "pinkTriangle": "#E91E63",
        "primary": "#039842",
        "purpleHaze": "#3F51B5",
        "purplePills": "#9B59B6",
        "redRedWine": "#EF4423",
        "royalBlood": "#FF5000",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
        "yellowSubmarine": "#FFD426",
      },
      "dimensions": Object {
        "button": Object {
          "default": Object {
            "height": "44px",
            "padding": "0 25px",
          },
          "small": Object {
            "height": "40px",
            "padding": "0 15px",
          },
        },
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
        "navbarHeight": Object {
          "lg": "84px",
          "md": "74px",
          "xs": "62px",
        },
        "radioBox": "20px",
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "formGroupLabel": "14px",
        "h1": Object {
          "lg": "48px",
          "md": "34px",
          "xs": "24px",
        },
        "h2": Object {
          "lg": "34px",
          "md": "24px",
          "xs": "20px",
        },
        "h3": Object {
          "lg": "24px",
          "md": "20px",
          "xs": "16px",
        },
        "h4": Object {
          "lg": "20px",
          "md": "16px",
          "xs": "16px",
        },
        "h5": Object {
          "lg": "16px",
          "md": "14px",
          "xs": "14px",
        },
        "h6": Object {
          "lg": "14px",
          "md": "14px",
          "xs": "14px",
        },
        "overline": "10px",
        "title": Object {
          "lg": "54px",
          "md": "38px",
          "xs": "26px",
        },
      },
      "fonts": Object {
        "regular": "Open Sans",
      },
      "media": Object {
        "lg": [Function],
        "md": [Function],
        "sm": [Function],
        "xl": [Function],
        "xs": [Function],
      },
      "ruler": Array [
        0,
        4,
        8,
        12,
        16,
        20,
        24,
        28,
        32,
        36,
        40,
        44,
        48,
        52,
        56,
        60,
        64,
        68,
        72,
        76,
        80,
      ],
      "utils": Object {
        "transition": [Function],
      },
      "zIndex": Object {
        "navbar": 100,
      },
    }
  }
>
  <styled.div>
    <div
      className="c0"
    >
      1
    </div>
  </styled.div>
</Component>
`;

exports[`<Badge /> Primary Badge renders the badge correctly 1`] = `
.c0 {
  width: 36px;
  height: 36px;
  margin: 0;
  padding: 0;
  font-size: 14px;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  border: none;
  border-radius: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: #039842;
  color: #FFFFFF;
  font-weight: 600;
}

<Component
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "default": "2px",
        "input": "2px",
        "large": "8px",
      },
      "colors": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueChristmas": "#3498DB",
        "blueMagic": "#34495E",
        "blueSmoke": "#949494",
        "brownSugar": "#9B525F",
        "checkBerry": "#98035A",
        "darkSmoke": "#767676",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green800": "#008736",
        "green900": "#006721",
        "greenGarden": "#19BC9B",
        "greenOnions": "#8BC34B",
        "macyGrey": "#DBDBDB",
        "mellowYellow": "#F1C410",
        "orangeCrush": "#F39C12",
        "paintItBlack": "#333333",
        "peaches": "#F99973",
        "pinkTriangle": "#E91E63",
        "primary": "#039842",
        "purpleHaze": "#3F51B5",
        "purplePills": "#9B59B6",
        "redRedWine": "#EF4423",
        "royalBlood": "#FF5000",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
        "yellowSubmarine": "#FFD426",
      },
      "dimensions": Object {
        "button": Object {
          "default": Object {
            "height": "44px",
            "padding": "0 25px",
          },
          "small": Object {
            "height": "40px",
            "padding": "0 15px",
          },
        },
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
        "navbarHeight": Object {
          "lg": "84px",
          "md": "74px",
          "xs": "62px",
        },
        "radioBox": "20px",
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "formGroupLabel": "14px",
        "h1": Object {
          "lg": "48px",
          "md": "34px",
          "xs": "24px",
        },
        "h2": Object {
          "lg": "34px",
          "md": "24px",
          "xs": "20px",
        },
        "h3": Object {
          "lg": "24px",
          "md": "20px",
          "xs": "16px",
        },
        "h4": Object {
          "lg": "20px",
          "md": "16px",
          "xs": "16px",
        },
        "h5": Object {
          "lg": "16px",
          "md": "14px",
          "xs": "14px",
        },
        "h6": Object {
          "lg": "14px",
          "md": "14px",
          "xs": "14px",
        },
        "overline": "10px",
        "title": Object {
          "lg": "54px",
          "md": "38px",
          "xs": "26px",
        },
      },
      "fonts": Object {
        "regular": "Open Sans",
      },
      "media": Object {
        "lg": [Function],
        "md": [Function],
        "sm": [Function],
        "xl": [Function],
        "xs": [Function],
      },
      "ruler": Array [
        0,
        4,
        8,
        12,
        16,
        20,
        24,
        28,
        32,
        36,
        40,
        44,
        48,
        52,
        56,
        60,
        64,
        68,
        72,
        76,
        80,
      ],
      "utils": Object {
        "transition": [Function],
      },
      "zIndex": Object {
        "navbar": 100,
      },
    }
  }
>
  <styled.div
    color="primary"
  >
    <div
      className="c0"
      color="primary"
    >
      1
    </div>
  </styled.div>
</Component>
`;
